scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -1047,6 +1047,6 @@ def weighted_fun(a, **kwargs):
     np.testing.assert_array_equal(res2, res)
     np.testing.assert_array_equal(res3, res)
     np.testing.assert_array_equal(res4, res)
-    if weighted_fun_name not in {'pmean'}:
+    if weighted_fun_name not in {'pmean', 'gmean'}:
         # _no_deco mean returns masked array, last element was masked
         np.testing.assert_allclose(res5.compressed(), res[~np.isnan(res)])
```

Comment:
`_no_deco` means "disable the `_axis_nan_policy` decorator". It's a private thing, just for testing. Here, we were using it to compare the behavior of the decorated `gmean`'s treatment of masked arrays against `gmean`'s internal treatment of masked arrays. Now that we're removing special masked array logic from the internals of `gmean`, we have to remove this part of the test.